contrib/mesos/pkg/scheduler/service/validation_test.go

```diff
@@ -0,0 +1,160 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/kubernetes/contrib/mesos/pkg/podutil"
+	mresource "k8s.io/kubernetes/contrib/mesos/pkg/scheduler/resource"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+)
+
+func TestStaticPodValidator(t *testing.T) {
+	// test within limits
+	tests := []struct {
+		// given
+		pods <-chan *api.Pod
+		// wants
+		podcount int
+		cputot   float64
+		memtot   float64
+	}{
+		// test: valid, pod specifies limits for ALL containers
+		{
+			pods: _pods(_pod(
+				_podName("foo", "bar"),
+				_containers(
+					_container(_resourceLimits(10, 20)), // min is 32
+					_container(_resourceLimits(30, 40)),
+					_container(_resourceLimits(50, 60)),
+				),
+			)),
+			podcount: 1,
+			cputot:   90,
+			memtot:   132,
+		},
+		// test: valid, multiple pods, specify limits for ALL containers
+		{
+			pods: _pods(
+				_pod(
+					_podName("foo", "bar"),
+					_containers(
+						_container(_resourceLimits(10, 20)), // min is 32
+						_container(_resourceLimits(30, 40)),
+						_container(_resourceLimits(50, 60)),
+					),
+				),
+				_pod(
+					_podName("kjh", "jkk"),
+					_containers(
+						_container(_resourceLimits(15, 25)), // min is 32
+						_container(_resourceLimits(35, 45)),
+						_container(_resourceLimits(55, 65)),
+					),
+				),
+			),
+			podcount: 2,
+			cputot:   195,
+			memtot:   274,
+		},
+		// test: no limits on CT in first pod so it's rejected
+		{
+			pods: _pods(
+				_pod(
+					_podName("foo", "bar"),
+					_containers(
+						_container(_resourceLimits(10, 20)), // min is 32
+						_container(),                        // min is 0.01, 32
+						_container(_resourceLimits(50, 60)),
+					),
+				),
+				_pod(
+					_podName("wza", "wer"),
+					_containers(
+						_container(_resourceLimits(10, 20)), // min is 32
+						_container(_resourceLimits(30, 40)),
+						_container(_resourceLimits(50, 60)),
+					),
+				),
+			),
+			podcount: 2,
+			cputot:   60.01 + 90,
+			memtot:   124 + 132,
+		},
+	}
+	for i, tc := range tests {
+		var cpu, mem float64
+		f := staticPodValidator(0, 0, &cpu, &mem)
+		list := podutil.List(f.Do(tc.pods))
+		assert.Equal(t, tc.podcount, len(list.Items), "test case #%d: expected %d pods instead of %d", i, tc.podcount, len(list.Items))
+		assert.EqualValues(t, tc.cputot, cpu, "test case #%d: expected %f total cpu instead of %f", i, tc.cputot, cpu)
+		assert.EqualValues(t, tc.memtot, mem, "test case #%d: expected %f total mem instead of %f", i, tc.memtot, mem)
+	}
+}
+
+type podOpt func(*api.Pod)
+type ctOpt func(*api.Container)
+
+func _pods(pods ...*api.Pod) <-chan *api.Pod {
```

Comment:
because I wanted short, easily readable names that I could be certain wouldn't be mistaken for anything other than test funcs. i'll rename if it hurts.